src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -0,0 +1,332 @@
+    public static final String CONNECTOR_READER_READER_GROUP_SCOPE = "connector.reader.reader-group.scope";
+    public static final String CONNECTOR_READER_READER_GROUP_NAME = "connector.reader.reader-group.name";
+    public static final String CONNECTOR_READER_READER_GROUP_REFRESH_INTERVAL = "connector.reader.reader-group.refresh-interval";
+    public static final String CONNECTOR_READER_READER_GROUP_EVENT_READ_TIMEOUT_INTERVAL = "connector.reader.reader-group.event-read-timeout-interval";
+    public static final String CONNECTOR_READER_READER_GROUP_CHECKPOINT_INITIATE_TIMEOUT_INTERVAL = "connector.reader.reader-group.checkpoint-initiate-timeout-interval";
+
+    // Writer Configurations
+    public static final String CONNECTOR_WRITER = "connector.writer";
+    public static final String CONNECTOR_WRITER_SCOPE = "connector.writer.scope";
+    public static final String CONNECTOR_WRITER_STREAM = "connector.writer.stream";
+    public static final String CONNECTOR_WRITER_MODE = "connector.writer.mode";
+    public static final String CONNECTOR_WRITER_MODE_VALUE_EXACTLY_ONCE = "exactly_once";
+    public static final String CONNECTOR_WRITER_MODE_VALUE_ATLEAST_ONCE = "atleast_once";
+    public static final String CONNECTOR_WRITER_TXN_LEASE_RENEWAL_INTERVAL = "connector.writer.txn-lease-renewal-interval";
+    public static final String CONNECTOR_WRITER_ROUTING_KEY_FILED_NAME = "connector.writer.routingkey-field-name";
+
+    private TableSourceReaderBuilder tableSourceReaderBuilder = null;
+
+    private TableSinkWriterBuilder tableSinkWriterBuilder = null;
+
+    public Pravega() {
+        super(CONNECTOR_TYPE_VALUE_PRAVEGA, CONNECTOR_VERSION_VALUE, true);
+    }
+
+    /**
+     * Internal method for connector properties conversion.
+     */
+    @Override
+    public void addConnectorProperties(DescriptorProperties properties) {
+        properties.putString(CONNECTOR_VERSION(), String.valueOf(CONNECTOR_VERSION_VALUE));
+
+        if (tableSourceReaderBuilder == null && tableSinkWriterBuilder == null) {
+            throw new ValidationException("Missing both reader and writer configurations.");
+        }
+
+        PravegaConfig pravegaConfig = tableSourceReaderBuilder != null ?
+                tableSourceReaderBuilder.getPravegaConfig() : tableSinkWriterBuilder.getPravegaConfig();
+        populateConnectionConfig(pravegaConfig, properties);
+
+        boolean metrics = tableSourceReaderBuilder != null ?
+                tableSourceReaderBuilder.isMetricsEnabled() : tableSinkWriterBuilder.isMetricsEnabled();
+        properties.putBoolean(CONNECTOR_METRICS, metrics);
+
+        if (tableSourceReaderBuilder != null) {
+            populateReaderProperties(properties);
+        }
+        if (tableSinkWriterBuilder != null) {
+            populateWriterProperties(properties);
+        }
+    }
+
+    /**
+     * Prepare Pravega connection specific configurations
+     * @param pravegaConfig
```

Comment:
If you add the parameters in the method description, it would be great to describe them in all the public methods.

Reply:
fixed